node/tests/swamp/swamp.go

```diff
@@ -296,3 +297,14 @@ func (s *Swamp) remove(rn *node.Node, sn []*node.Node) ([]*node.Node, error) {
 	}
 	return sn, nil
 }
+
+// Disconnect allows to break a connection between two peers without any possibility to re-establish it.
+// Order is very important here. We have to unlink peers first, and only after that call disconnect,
+// otherwise pubsub could re-establish the connection.
+func (s *Swamp) Disconnect(peerA, peerB peer.ID) error {
+	err := s.Network.UnlinkPeers(peerA, peerB)
+	if err != nil {
+		return err
+	}
+	return s.Network.DisconnectPeers(peerA, peerB)
+}
```

Comment:
Would be better to make rcv the T, so it checks for errors inside. This just generally makes tests cleaner.